modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/ConfigInitializerContextListener.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.common.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.w3c.dom.Document;
+import org.wso2.carbon.mdm.mobileservices.windows.common.Constants;
+import org.xml.sax.SAXException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+
+import org.wso2.carbon.mdm.mobileservices.windows.common.beans.WindowsPluginProperties;
+
+/**
+ * This class performs one time operations.
+ */
+public class ConfigInitializerContextListener implements ServletContextListener {
+
+    public static final int INITIAL_VALUE = 0;
+    private static Log log = LogFactory.getLog(ConfigInitializerContextListener.class);
+    private enum PropertyName {
```

Comment:
Please use a meaningful name for the enum declaration. If the attributes nested within the enum refers to something like properties related to certificates, etc, maybe "CertificateProperties" makes a good name for it.